SPEC-annex-A-filesystem-storage.md

```diff
@@ -0,0 +1,49 @@
+## Annex A
+
+Annex A documents known methods of persisting GitBOM Documents to various stores.
+### GitBOM Document persistence by a Build Tool to its local filesystem
+
+If a build tool persists GitBOM information to its local filesystem, the build tool should write out the GitBOM Document to ```${GITBOM_DIR}/objects/${Artifact Identifier Type uri prefix with ':' replaced by '/'}/${GitBOMID:0:2}/${GitBOMID:2:}``` where ```${GitBOMID}``` is the GitBOM id in lowercase hexidecimal without leading zeros NOT suppressed.
+
+Example:
+
+```
+.adg/objects/gitoid/blob/sha1/0e/8efd4cdf0d5bafcfcae658c2662a73b199b301
+```
+
+#### Build tool persistence of artifact to GitBOM Document mapping
+
+A build tool should choose to persist a mapping between artifacts and their corresponding GitBOM documents.  If it chooses
+to do so it should for each artifact persist a symlink:
+
+```${GITBOM_DIR}/a2g/${Artifact Identifier Type uri prefix with ':' replaced by '/'}/${artifact id}:0:2}/${artifact id:2:} -> ${GITBOM_DIR}/objects/${Artifact Identifier Type uri prefix with ':' replaced by '/'}/${GitBOMID:0:2}/${GitBOMID:2:}```
+
+`a2g` is short for `artifact to graph`.
+
+Example:
+
+```
+.adg/a2g/gitoid/blob/sha1/0e/8efd4cdf0d5bafcfcae658c2662a73b199b301 -> .adg/objects/gitoid/blob/sha1/1d/6e79da5e380e5d3e5adcf899c4d65c0e80bfb3
+```
+
+#### Build tool persistence of related metadata
+
+A build tool may persist additional metadata to that makes reference to the Artifact Dependency Graph (ADG).
+It should persist such metadata to a subdirectory of the directory to which the output artifact is being written of the form: ```${GITBOM_DIR}/metadata/${context}/```.  
+
+For metadata specific to a particular build tool ```${context}``` should be a name uniquely associated with the build tool.  Build tools should report their selection of ```${context}``` subdirectory name to the GitBOM spec for inclusion in a list to preclude ```${context}``` collision.
```

Comment:
> Build tools should report their selection

That's build tool authors/developers, right? Not the build tool itself doing something at its runtime?

Reply:
Proposed fix in https://github.com/git-bom/spec/pull/20/commits/4d4648c74cfeee458c7e3634f6b22591a0cc17cf

Reply:
Fixed in https://github.com/git-bom/spec/pull/20/commits/b00a20c476de9b0fde4f1c28d07191a620602fdc